### DISCUSSION TITLE
Validate canary without taking traffic

### DISCUSSION THREAD

**Discussion by Bennett-Lynch** [argoproj/argo-rollouts#3890] [Q&A]
Hello,

How can I run an analysis against a canary replica set before increasing its traffic weight, when I'm not using traffic routing/traffic management? Is this supported?

Thanks!

**Comment by kostis-codefresh** [comment]
Yes this is supported

Read the documentation about the difference between setCanaryScale and setWeight

if, for example, you do setCanaryScale = 3, you will have 3 pods that don't get any traffic (setWeight=0)

Then you can run any smoke tests that you want with https://argo-rollouts.readthedocs.io/en/stable/analysis/job/ or https://argo-rollouts.readthedocs.io/en/stable/analysis/web/

**Reply by Bennett-Lynch** [reply]
Hi @kostis-codefresh. According to the docs:

> Setting canary scale is only available when using the canary strategy with a traffic router, since the basic canary needs to control canary scale in order to approximate canary weight.

https://argo-rollouts.readthedocs.io/en/stable/features/canary/#dynamic-canary-scale-with-traffic-routing

My question is: how can I do this _(test the canary replica set w/o it receiving traffic)_ when NOT using a traffic router? I seem to recall @jessesuen mentioning this was possible somewhere but I cannot find it now.
